File: Two Pointers/Problem2 - 392.java
public class Main {
    /* Optimal Approach - using two pointer approach */
    public static boolean isSubsequence(String s, String t) {
        int n = s.length(), m = t.length();
        if (n > m) return false;

        int i = 0, j = 0;
        while (i < n && j < m) {
            if (s.charAt(i) == t.charAt(j)) {
                i++;
            }
            j++;
        }
        return i == n;
        /*  TC-O(n + n) SC-O(1) */
    }

    public static void main(String[] args) {
        String s = "abc";
        String t = "ahbgdc";
        System.out.println(isSubsequence(s, t)); // Output: true
    }
}

File: Two Pointers/Problem2 - 392.cpp
#include <bits/stdc++.h>
using namespace std;

/* Optimal Approach - using two pointer approach */
bool isSubsequence(string s, string t) {
    int n = s.size(), m = t.size();
    if (n > m) return false;

    int i = 0, j = 0;
    while (i < n && j < m) {
        if (s[i] == t[j])  i++;
        j++;
    }
    return i == n;
    /*  TC-O(n) SC-O(1) */
}